File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <map>

using namespace std;

map<string, int> m;

vector<int> solution(int n, vector<string> words) {
    int now = 1;
    
    m[words[0]] = 1;
    char prev = words[0].back();
    
    for(int i = 1; i < words.size(); i++)
    {
        if(prev != words[i].front()) return {now + 1, i/n + 1};
        if(m.count(words[i]) == 0) m[words[i]] = 1;
        else
        {
            return {now + 1, i/n + 1};
        }
        now = (now + 1) % n;
        prev = words[i].back();
    }        
    
    return {0, 0};
    
}

File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.cs
using System;
using System.Collections.Generic;

class Solution
{
    public int[] solution(int n, string[] words)
    {
        int[] answer = new int[2] {0,0};
        
        Dictionary<string, int> m = new Dictionary<string, int>();
        m.Add(words[0], 1);

        int now = 1;
        char prev = words[0][words[0].Length -1];
        
        for(int i = 1; i < words.Length; i++)
        {
            if(prev != words[i][0]) 
            {
                answer[0] = now+1; 
                answer[1] = i/n + 1;
                return answer; 
            }
            if(!m.ContainsKey(words[i])) m.Add(words[i], 1);
            else
            {
                answer[0] = now + 1;
                answer[1] = i/n + 1;
                return answer;
            }
            now = (now + 1) % n;
            prev = words[i][words[i].Length - 1];
        }
        
        return answer;
    }
}